src/main/java/ca/uwaterloo/iqc/topchef/endpoint_models/JSONSchemaValidator.java

```diff
@@ -0,0 +1,172 @@
+package ca.uwaterloo.iqc.topchef.endpoint_models;
+
+import ca.uwaterloo.iqc.topchef.Client;
+import ca.uwaterloo.iqc.topchef.adapters.java.net.HTTPRequestMethod;
+import ca.uwaterloo.iqc.topchef.adapters.java.net.HTTPResponseCode;
+import ca.uwaterloo.iqc.topchef.adapters.java.net.URL;
+import ca.uwaterloo.iqc.topchef.adapters.java.net.URLConnection;
+import ca.uwaterloo.iqc.topchef.exceptions.HTTPConnectionCastException;
+import ca.uwaterloo.iqc.topchef.exceptions.UnexpectedResponseCodeException;
+import org.json.simple.JSONAware;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+
+/**
+ * Endpoint for validating URLs
+ */
+public class JSONSchemaValidator extends AbstractEndpoint implements Validator {
+
+    /**
+     * The log to which runtime messages are written
+     */
+    private static final Logger log = LoggerFactory.getLogger(JSONSchemaValidator.class);
+
+    /**
+     *
+     * @param url The URL at which the validator is located
+     */
+    public JSONSchemaValidator(URL url){
+        super(url);
+    }
+
+    /**
+     *
+     * @param client The client that this validator is being run from. The validator uses the client's URL resolver to
+     *               find where the validator is
+     */
+    public JSONSchemaValidator(Client client){
+        super(client.getURLResolver().getValidatorEndpoint());
+    }
+
+    /**
+     *
+     * @param instance The JSON object to be validated
+     * @param schema The schema against which the object is to be validated
+     * @return {@link Boolean#TRUE} if the instance matches the schema, otherwise {@link Boolean#FALSE}
+     * @throws IOException If I/O cannot be established with the TopChef API
+     * @throws UnexpectedResponseCodeException if the response code is not {@link HTTPResponseCode#OK} and
+     * {@link HTTPResponseCode#BAD_REQUEST}
+     */
+    @Override
+    public Boolean validate(JSONAware instance, JSONAware schema) throws IOException, UnexpectedResponseCodeException {
```

Comment:
I don't think so, because it's not the same definition. ``getDataToSend`` is called with different types. In order to have it be the same definition, I would have to define a new type of which ``JSONAware`` and ``String`` would be instances, and to do that, I would have to encapsulate ``String`` and give it a ``toJSONString`` method.